### PR TITLE
Move to food inspection data

### PR DESCRIPTION
**Pull request by isabelizimm**
I think I've moved over to the new data! A few todos around reading in the data from `board_connect_url`, but it should be mostly there. It certainly is not as clean of a dataset as the abalone, but am pretty amused at with...one restaurant has 400+ violations 😱

### CODE REVIEW

File: slides/01-introduction.qmd [hunk 1]
```diff
@@ -324,9 +340,9 @@ abalone |>
 #| dev-args: list(bg = "transparent")
 tree_mod <- 
   rpart::rpart(
-    rings ~ .,
-    data = abalone,
-    control = rpart::rpart.control(maxdepth = 3, cp = 0.001)
+    results ~ facility_type + risk + total_violations,
+    data = inspection,
+    #control = rpart::rpart.control(maxdepth = 3, cp = 0.001)
```
Comment (isabelizimm) [inline slides/01-introduction.qmd:345]: not sure if the control is still relevant...this plot is not as useful with this dataset

File: slides/01-introduction.qmd [hunk 2]
```diff
@@ -351,7 +367,7 @@ ring_test <- testing(ring_split)
 from sklearn import model_selection
 import numpy as np
 np.random.seed(123)
-X, y = abalone.iloc[:,:8], abalone['rings']
+X, y = inspection.drop(columns=['aka_name', 'inspection_date', 'results']), inspection['results']
```
Comment (isabelizimm) [inline slides/01-introduction.qmd:370]: I can also drop the name and inspection date from the pinned data, but maybe we can use the inspection date to talk about monitoring?

File: slides/02-deploy.qmd [hunk 3]
```diff
@@ -45,40 +45,56 @@ library(vetiver)
...
-abalone <- read_csv("abalone.csv")
+board <- board_connect_url(c(
+  inspection_data = 'https://colorado.posit.co/rsc/inspection_data/'
+  ))
+
+inspection <- read_csv('/Users/isabelzimmerman/code/github/vetiver/slides/inspection.csv') 
+# TODO: read pin from vanity url
+# board %>% pin_read("inspection_data")
+
+# TODO: can we set str to factors?
```
Comment (isabelizimm) [inline slides/02-deploy.qmd:56]: same TODOs as the other file, this is all just copied over.

File: slides/01-introduction.qmd (hunk 1 above)
Comment (juliasilge) [reply slides/01-introduction.qmd]: I got this to give us a nice-ish plot (nice enough to talk over about training a model) 👍

File: slides/01-introduction.qmd [hunk 4]
```diff
@@ -234,45 +251,36 @@ abalone <- read_csv("abalone.csv")
 
 ```{python}
 import pandas as pd
-abalone = pd.read_csv("abalone.csv")  
+pd.read_parquet('./data/inspections.parquet', engine = 'pyarrow')
```
Comment (juliasilge) [inline slides/01-introduction.qmd:254]: Is this right @isabelizimm?

File: slides/01-introduction.qmd [hunk 5]
```diff
@@ -351,36 +362,45 @@ ring_test <- testing(ring_split)
...
 ```
 
 ### Python
 
 ```{python}
-from sklearn import preprocessing, tree, pipeline
-le = preprocessing.OrdinalEncoder().fit(X_train)
-tr = tree.DecisionTreeRegressor().fit(le.transform(X_train), y_train)
-abalone_fit = pipeline.Pipeline([('label_encoder', le), ('decision_tree', tr)])
+## TODO for Isabel: can you change this to logistic regression and add the `inspection_date` predictor?
```
Comment (juliasilge) [inline slides/01-introduction.qmd:391]: Here is the first TODO for you @isabelizimm.

File: slides/02-deploy.qmd [hunk 6]
```diff
@@ -41,44 +41,53 @@ knitr:
...
+    rand_forest(mode = "classification", trees = 1e3)
   ) |> 
-  fit(data = ring_train) 
+  fit(data = inspect_train)
 ```
 
 ### Python
 
 ```{python}
+## TODO for Isabel: can you add the `inspection_date` predictor to this random forest model?
```
Comment (juliasilge) [inline slides/02-deploy.qmd:64]: Here is TODO 2 for you! 2️⃣

File: slides/02-deploy.qmd [hunk 7]
```diff
@@ -41,44 +41,53 @@ knitr:
...
+## TODO for Isabel: can you add a script to `data-raw` that trains this model and deploys to Connect? see my `prep_chicago_inspections.R` but *only* do model training, not data prep
```
Comment (juliasilge) [inline slides/02-deploy.qmd:65]: Here is TODO 3 for you 3️⃣ 

Let me know if this item doesn't make sense; you can add something like a `prep_chicago_inspections.py` to `data-raw` and keep a script to train a random forest and deploy it to Connect.

File: slides/02-deploy.qmd [hunk 8]
```diff
@@ -280,19 +289,18 @@ You can treat your model API much like it is a local model in memory!
...
+```{r}
+endpoint <- vetiver_endpoint("https://colorado.posit.co/rsc/chicago-inspections-rstats/predict")
+predict(endpoint, slice_sample(inspect_test, n = 10))
 ```
 
 ### Python
 
 ```python
-## TODO: put a model on Colorado with our dataset and run this for real
+## TODO for Isabel: put a model on Colorado using the script in `data-raw` and run this for real
```
Comment (juliasilge) [inline slides/02-deploy.qmd:300]: Here is TODO 4️⃣

File: slides/02-deploy.qmd [hunk 9]
```diff
@@ -330,15 +338,14 @@ What if your model API requires authentication?
...
-predict(endpoint, slice_sample(ring_test, n = 10))
+endpoint <- vetiver_endpoint("https://colorado.posit.co/rsc/chicago-inspections-rstats/predict")
+predict(endpoint, slice_sample(inspect_test, n = 10))
 ```
 
 ### Python
 
 ```python
-## TODO: put a model on Colorado with our dataset and run this for real
+## TODO for Isabel: update this with real URL but we will not run (just here for auto-animate)
```
Comment (juliasilge) [inline slides/02-deploy.qmd:348]: Here is TODO 5️⃣

File: slides/02-deploy.qmd [hunk 10]
```diff
@@ -351,20 +358,21 @@ What if your model API requires authentication?
...
 apiKey <- Sys.getenv("CONNECT_API_KEY")
-predict(endpoint, slice_sample(ring_test, n = 10), 
+predict(endpoint, slice_sample(inspect_test, n = 10), 
         httr::add_headers(Authorization = paste("Key", apiKey)))
 ```
 
 ### Python
 
 ```python
+## TODO for Isabel: update this with real URL but we will not run (just here for auto-animate)
```
Comment (juliasilge) [inline slides/02-deploy.qmd:370]: Here is TODO 6️⃣

**Review by juliasilge — COMMENTED** [review]
Thank you so much for all the work on this @isabelizimm! I outlined six specific things for you to address still here, and then we'll merge in and keep going.

File: slides/01-introduction.qmd (hunk 4 above)
Comment (isabelizimm) [reply slides/01-introduction.qmd:254]: Yep! I'll just make sure that pyarrow is in the appropriate requirements files

File: slides/02-deploy.qmd [hunk 11]
```diff
@@ -263,7 +272,9 @@ jsonlite::fromJSON(metadata)
 
 ```{python}
 #| eval: false
-board.pin_meta("isabel.zimmerman/test_model").user
+import requests
+url = "https://colorado.posit.co/rsc/chicago-inspections-python/metadata" 
+print(requests.get(url).content)
```
Comment (isabelizimm) [inline slides/02-deploy.qmd:277]: I did update this requests call, since I was going through anyway. When this PR gets merged, I'll close out #8

**Pull request merged**